mgradm/cmd/distro/cp.go

```
@@ -98,28 +74,98 @@ func distroCp(
 			srcdir,
 		}
 		if out, err := utils.RunCmdOutput(zerolog.DebugLevel, "/usr/bin/sudo", mountCmd...); err != nil {
-			log.Debug().Msgf("Error mounting ISO image: '%s'", out)
-			return errors.New(L("unable to mount ISO image. Mount manually and try again"))
+			log.Debug().Msgf(L("Error mounting ISO image: '%s'"), out)
+			return "", needremove, fmt.Errorf(L("unable to mount iso file: %s"), out)
 		}
+		needremove = true
 	}
+	return srcdir, needremove, nil
+}
+
+func copyDistro(srcdir string, distro types.Distribution, flags *flagpole) error {
+	cnx := shared.NewConnection(flags.Backend, podman.ServerContainerName, kubernetes.ServerFilter)
 
+	const distrosPath = "/srv/www/distributions/"
+	dstpath := distrosPath + distro.TreeLabel
+	distro.BasePath = dstpath
+	if cnx.TestExistenceInPod(dstpath) {
+		return fmt.Errorf(L("distribution with same name already exists: %s"), dstpath)
+	}
+
+	if _, err := cnx.Exec("sh", "-c", "mkdir -p "+distrosPath); err != nil {
+		return fmt.Errorf(L("cannot create %s path in container: %s"), distrosPath, err)
+	}
+
+	log.Info().Msgf(L("Copying distribution %s"), distro.TreeLabel)
 	if err := cnx.Copy(srcdir, "server:"+dstpath, "tomcat", "susemanager"); err != nil {
 		return fmt.Errorf(L("cannot copy %s: %s"), dstpath, err)
 	}
+	log.Info().Msgf(L("Distribution has been copied into %s"), distro.BasePath)
+	return nil
+}
 
-	log.Info().Msg(L("Distribution has been copied"))
+func getServerFqdn(flags *flagpole) (string, error) {
```

Shouldn't this be in the API module and exposed?

---

Perhaps. Issue is that api is on both `mgrctl` and `mgradm` but only `mgradm` we expect to be run on the host server.
I know we have remote podman connection, etc. but I consider them as optional, particularly for api calls.

I am going to move it to the api module and let it fail if `--api-host` or podman connection is not available.

---

> Perhaps. Issue is that api is on both `mgrctl` and `mgradm` but only `mgradm` we expect to be run on the host server. I know we have remote podman connection, etc. but I consider them as optional, particularly for api calls.

Oh right I forgot.

> I am going to move it to the api module and let it fail if `--api-host` or podman connection is not available.

No need, keep it as is, I forgot about `mgrctl` not having the file access. We'll implement https://github.com/uyuni-project/uyuni-tools/issues/239 instead for it.